clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/SimpleClusterTierClientEntity.java

```diff
@@ -209,6 +222,32 @@ public EhcacheEntityResponse invokeAndWaitForRetired(EhcacheOperationMessage mes
     return invokeInternalAndWait(endpoint.beginInvoke().blockGetOnRetire(true), message, track);
   }
 
+  private void waitForInitCompleted(long nanos) {
+    boolean interrupted = false;
+    long deadlineTimeout = System.nanoTime() + nanos;
+    try {
+      while (true) {
+        try {
+          long timeRemaining = deadlineTimeout - System.nanoTime();
+          if (timeRemaining <= 0 || initCompleted) {
+            return;
+          }
+          synchronized (initLock) {
+            if (!initCompleted) {
+              lock.wait(TimeUnit.NANOSECONDS.toMillis(timeRemaining));
+            }
+          }
+        } catch (InterruptedException e) {
+          interrupted = true;
```

Comment:
Not going out of the loop when interrupted?

Reply:
No. But I am restoring the interrupt state of the thread. Did not want such a crucial wait to be interrupted leaving the overall state in a mess..This way atleast we have a clear exit path on interrupt. Anyway it is not an indefinite wait...